sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/context/AuthTokenContext.java

```diff
@@ -83,23 +91,41 @@ public synchronized String getAuthToken() {
         return mAuthToken;
     }
 
+    /**
+     * Gets current homeAccountId value.
+     *
+     * @return unique identifier of user.
+     */
+    public synchronized String getHomeAccountId() {
+        return mHomeAccountId;
+    }
+
     /**
      * Sets new authorization token.
      *
      * @param authToken     authorization token.
      * @param homeAccountId unique user id.
+     * @param expiresOn     time when token expires.
      */
-    public synchronized void setAuthToken(String authToken, String homeAccountId) {
+    public synchronized void setAuthToken(String authToken, String homeAccountId, Date expiresOn) {
+        if (mStorage != null) {
+            mStorage.saveToken(authToken, homeAccountId, expiresOn);
+        }
+        updateAuthToken(authToken, homeAccountId);
+    }
+
+    private void updateAuthToken(String authToken, String homeAccountId) {
+        final boolean isNewUser = isNewUser(homeAccountId);
```

Comment:
```suggestion
        boolean isNewUser = isNewUser(homeAccountId);
```

Reply:
but why do we need a local variable now? It's used only once.

Reply:
1. It's used in the loop
2. We need to check in before changing `mHomeAccountId`

I'll add comment